docs/how-to/hip_runtime_api/memory_management/virtual_memory.rst

```diff
@@ -0,0 +1,154 @@
+    padded_size = ROUND_UP(size, granularity);
+    hipMemCreate(&allocHandle, padded_size, &prop, 0);
+
+Reserve virtual address range
+--------------------------------------------------------------------------------
+
+After you have acquired an allocation of physical memory, you must map it before
+you can use it. To do so, you need a virtual address to map it to.  Mapping
+means the physical memory allocation is available from the virtual address range
+it is mapped to. To reserve a virtual memory range, use the
+:cpp:func:`hipMemAddressReserve` function. The size of the virtual memory must
+match the amount of physical memory previously allocated. You can then map the
+physical memory allocation to the newly-acquired virtual memory address range
+using the :cpp:func:`hipMemMap` function.
+
+.. code-block:: cpp
+
+    hipMemAddressReserve(&ptr, padded_size, 0, 0, 0);
+    hipMemMap(ptr, padded_size, 0, allocHandle, 0);
+
+Set memory access
+--------------------------------------------------------------------------------
+
+Finally, use the :cpp:func:`hipMemSetAccess` function to enable memory access.
+It accepts the pointer to the virtual memory, the size, and a
+:cpp:struct:`hipMemAccessDesc` descriptor as parameters. In a multi-GPU
+environment, you can map the device memory of one GPU to another. This feature
+also works with the traditional memory management system, but isn't as scalable
+as with virtual memory. When memory is allocated with :cpp:func:`hipMalloc`,
+:cpp:func:`hipDeviceEnablePeerAccess` is used to enable peer access. This
+function enables access between two devices, but it means that every call to
+:cpp:func:`hipMalloc` takes more time to perform the checks and the mapping
+between the devices. When using virtual memory management, peer access is
+enabled by :cpp:func:`hipMemSetAccess`, which provides a finer level of
+control over what is shared. This has no performance impact on memory allocation
+and gives you more control over what memory buffers are shared with which
+devices.
+
+.. code-block:: cpp
+
+    hipMemAccessDesc accessDesc = {};
+    accessDesc.location.type = HIP_MEM_LOCATION_TYPE_DEVICE;
+    accessDesc.location.id = currentDev;
+    accessDesc.flags = HIP_MEM_ACCESS_FLAGS_PROT_READWRITE;
+    hipMemSetAccess(ptr, padded_size, &accessDesc, 1);
+
+At this point the memory is allocated, mapped, and ready for use. You can read
+and write to it, just like you would a C style memory allocation.
+
+Free virtual memory
+--------------------------------------------------------------------------------
+
+To free the memory allocated in this manner, use the corresponding free
+functions. To unmap the memory, use :cpp:func:`hipMemUnmap`. To release the
+virtual address range, use :cpp:func:`hipMemAddressFree`.  Finally, to release
+the physical memory, use :cpp:func:`hipMemRelease`. A side effect of these
+functions is the lack of synchronization when memory is released. If you call
```

Comment:
If you call :cpp:func:`hipFree` when you have multiple streams running in parallel, it
synchronizes the device. This causes worse resource usage and performance.

This seems like it could be an important note?